example/builder/src/new_class_consumed_props.dart

```diff
@@ -0,0 +1,59 @@
+// Copyright 2020 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:over_react/over_react.dart';
+
+// ignore_for_file: uri_has_not_been_generated
+part 'new_class_consumed_props.over_react.g.dart';
+
+UiFactory<SomeParentProps> SomeClassParent = _$SomeClassParent; // ignore: undefined_identifier
+
+mixin ParentOnlyPropsMixin on UiProps {
+  String aParentProp;
+}
+
+mixin SharedPropsMixin on UiProps {
+  String aPropToBePassed;
+}
+
+class SomeParentProps = UiProps with ParentOnlyPropsMixin, SharedPropsMixin;
+
+class SomeClassParentComponent extends UiComponent2<SomeParentProps> {
+  @override
+  render() {
+    UiPropsMeta(props).meta.consume(ParentOnlyPropsMixin);
```

Comment:
I didn't want to bring this up for discussion until after the APIs are a little more solid, but here I was wondering: could this replace the original `consumedProps` mechanism? It would keep the API consistent and reduce some of what the builder needs to do, since this is available on all `Component2` instances anyway. 

The main tradeoff is that this system isn't as intuitive for class based components and is really only manageable if they pass along props in `render` since they need to be re-consume the props every time a new props instance is received (which also has a performance implication, even if slight)